### ISSUE TITLE
no-empty-interface false positive when interface extends another with generics

### ISSUE THREAD

**Issue by nicocevallos** [typescript-eslint/typescript-eslint#375]
**Repro**
* Use the recommended configuration
* Create a interface with generics
* Create a second interface which extends from the first one, but setting the values for generics

```JSON
{
  "rules": {
    "@typescript-eslint/no-empty-interface": [
      "error",
      {
        "allowSingleExtends": false,
      }
    ]
  }
}
```

```TS
interface A<T>{
  t: T;
}

interface B extends A<number>{
}
```

**Expected Result**
`no-empty-interface` rule should not be triggered for `SecondInterface` interface

**Actual Result**
`no-empty-interface` is triggered unless you set the `allowSingleExtends` option to `true`

**Additional Info**
I think setting `allowSingleExtends` to `true` should not be the solution becaus `interface B extends A<T> {}` is not the same as `interface B extends A {}`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.5.0` |
| `@typescript-eslint/parser`        | `1.5.0` |
| `TypeScript`                       | `3.3.4000` |
| `ESLint`                           | `5.15.3` |
| `node`                             | `10.15.3` |
| `npm`                              | `6.4.1` |
| `yarn`                             | `1.15.2` |

**Comment by bradzacher** [comment]
This is working as intended.
If you want to extend a generic interface only to pick a generic you can use a type statement:
```TS
type B = A<number>
```

Duplicate of #351
